src/cli/cmd_clean.py

```diff
@@ -0,0 +1,150 @@
+# pylint: disable=duplicate-code
+"""Module for the cleaning command."""
+import logging
+import sys
+from datetime import datetime
+from pathlib import Path
+from typing import Annotated, Optional
+
+import typer
+
+from src.blueprints.datafile import RawDatafile
+from src.cli.common import (
+    LogFileOption,
+    ProfileNameOption,
+    ProfileVersionOption,
+    SourceDataPathArg,
+    StorageBoxOption,
+    get_config,
+)
+from src.config.config import ConfigFromEnv
+from src.inspector.inspector import Inspector
+from src.profiles.profile_register import load_profile
+from src.utils import log_utils
+from src.utils.filesystem.ctime import max_ctime
+from src.utils.timing import Timer
+
+logger = logging.getLogger(__name__)
+
+
+def _check_verified_status(config: ConfigFromEnv, datafiles: list[RawDatafile]) -> None:
+    logger.info("Retrieving status...")
+    # Check ingestion status for each file.
+    timer = Timer(start=True)
+    insepctor = Inspector(config)
+    results = [insepctor.is_datafile_verified(raw_df) for raw_df in datafiles]
+    logger.info("Retrieved datafile status in %f seconds.", timer.stop())
+    # Get unverified files by the index of the verification results list.
+    unverified_dfs = [
+        datafiles[ind] for ind, is_verified in enumerate(results) if not is_verified
+    ]
+
+    if len(unverified_dfs) > 0:
```

Comment:
Is it worth refactoring to pass back a list of verified files that can be deleted, or do you feel that this adds additional complexity without much benefit?